python/pyspark/mllib/linalg.py

```diff
@@ -670,6 +674,18 @@ def toArray(self):
         """
         return self.values.reshape((self.numRows, self.numCols), order='F')
 
+    def toSparse(self):
+        """Convert to SparseMatrix"""
+        indices = np.nonzero(self.values)[0]
+        colCount = np.bincount(indices / self.numRows + 1)
```

Comment:
On second thought, this might be more readable:

~~~python
colCounts = np.bincount(indices / self.numRows)
colPtrs = np.cumsum(np.hstack((0, colCounts, np.zeros(self.numCols - colCounts.size))))
~~~